CI-CD is ok, just some problems with github action:
```
2024-02-21T08:52:17.7500878Z в„№пёЏ - Java CI / Test Report (17) - 15379 tests run, 15279 passed, 100 skipped, 0 failed.
2024-02-21T08:52:17.7516214Z в„№пёЏ - Java CI / Test Report (17) - Creating check (Annotations: 0)
2024-02-21T08:52:18.0176377Z ##[error]вќЊ Failed to create checks using the provided token. (HttpError: Resource not accessible by integration)
2024-02-21T08:52:18.0180259Z ##[warning]вљ пёЏ This usually indicates insufficient permissions. More details: https://github.com/mikepenz/action-junit-report/issues/23
```

--- a/core-processor/src/main/java/io/micronaut/inject/configuration/ConfigurationUtils.java
+++ b/core-processor/src/main/java/io/micronaut/inject/configuration/ConfigurationUtils.java
@@ -48,7 +48,7 @@ public final class ConfigurationUtils {
         } else {
             typePath = getRequiredTypePath(owningType);
         }
-        return typePath + "." + propertyName;
+        return typePath + '.' + propertyName;
     }
 
     public static String getRequiredTypePath(ClassElement classElement) {
@@ -57,7 +57,7 @@ public final class ConfigurationUtils {
 
     public static Optional<String> getTypePath(ClassElement classElement) {
         if (!classElement.hasStereotype(ConfigurationReader.class)) {
-            return Optional.empty();
+            return Optional.of(StringUtils.EMPTY_STRING);
         }
         if (classElement.isTrue(ConfigurationReader.class, ConfigurationReader.PREFIX_CALCULATED)) {
             return classElement.stringValue(ConfigurationReader.class, ConfigurationReader.PREFIX);
@@ -75,7 +75,7 @@ public final class ConfigurationUtils {
 
     private static String combinePaths(String p1, String p2) {
         if (StringUtils.isNotEmpty(p1) && StringUtils.isNotEmpty(p2)) {
-            return p1 + "." + p2;
+            return p1 + '.' + p2;
         }
         if (StringUtils.isNotEmpty(p1)) {
             return p1;
@@ -91,7 +91,7 @@ public final class ConfigurationUtils {
             if (prefixOptional.isEmpty()) {
                 prefix = basePrefixOptional.get();
             } else {
-                prefix = prefixOptional.map(p -> basePrefixOptional.get() + "." + p).orElse(null);
+                prefix = prefixOptional.map(p -> basePrefixOptional.get() + '.' + p).orElse(null);
             }
         } else {
             prefix = prefixOptional.orElse(null);
@@ -100,7 +100,7 @@ public final class ConfigurationUtils {
             return computeIterablePrefix(annotationMetadata, prefix);
         }
         if (prefix == null) {
-            return "";
+            return StringUtils.EMPTY_STRING;
         }
         return prefix;
     }
@@ -128,7 +128,8 @@ public final class ConfigurationUtils {
         while (classElement.isInner() && inner.isPresent()) {
             ClassElement enclosingType = inner.get();
             if (enclosingType.isTrue(ConfigurationReader.class, ConfigurationReader.PREFIX_CALCULATED)) {
-                String parentPrefix = enclosingType.stringValue(ConfigurationReader.class, ConfigurationReader.PREFIX).orElse("");
+                String parentPrefix = enclosingType.stringValue(ConfigurationReader.class, ConfigurationReader.PREFIX)
+                        .orElse(StringUtils.EMPTY_STRING);
                 path = combinePaths(parentPrefix, path);
                 break;
             } else {
@@ -149,7 +150,8 @@ public final class ConfigurationUtils {
             while (optionalSuperType.isPresent()) {
                 ClassElement superType = optionalSuperType.get();
                 if (superType.isTrue(ConfigurationReader.class, ConfigurationReader.PREFIX_CALCULATED)) {
-                    String parentPrefix = superType.stringValue(ConfigurationReader.class, ConfigurationReader.PREFIX).orElse("");
+                    String parentPrefix = superType.stringValue(ConfigurationReader.class, ConfigurationReader.PREFIX)
+                            .orElse(StringUtils.EMPTY_STRING);
                     path = combinePaths(parentPrefix, path);
                     break;
                 } else {
@@ -177,7 +179,10 @@ public final class ConfigurationUtils {
     }
 
     private static ClassElement resolveSuperInterface(ClassElement declaringType) {
-        return declaringType.getInterfaces().stream().filter(tm -> tm.hasStereotype(ConfigurationReader.class)).findFirst().orElse(null);
+        return declaringType.getInterfaces().stream()
+                .filter(tm -> tm.hasStereotype(ConfigurationReader.class))
+                .findFirst()
+                .orElse(null);
     }
 
 }

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/AbstractConfig.java
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/AbstractConfig.java
@@ -1,0 +1,32 @@
+package io.micronaut.inject.configproperties.inheritance;
+
+import org.jetbrains.annotations.NotNull;
+
+public abstract class AbstractConfig {
+    @NotNull
+    private final String value;
+    @NotNull
+    private String notThing;
+
+    public AbstractConfig(@NotNull String value) {
+        this.value = value;
+        this.notThing = "def notThing";
+    }
+
+    @NotNull
+    public final String getValue() {
+        return this.value;
+    }
+
+    @NotNull
+    public final String getNotThing() {
+        return this.notThing;
+    }
+
+    public final void setNotThing(@NotNull String notThing) {
+        this.notThing = notThing;
+    }
+
+    @NotNull
+    public abstract String getThing();
+}

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/AbstractConfigImpl.java
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/AbstractConfigImpl.java
@@ -1,0 +1,49 @@
+package io.micronaut.inject.configproperties.inheritance;
+
+import io.micronaut.context.annotation.ConfigurationInject;
+import io.micronaut.context.annotation.EachProperty;
+import io.micronaut.context.annotation.Parameter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@EachProperty("teams")
+public class AbstractConfigImpl extends AbstractConfig {
+    @NotNull
+    private final String name;
+    @Nullable
+    private String childThing;
+    @NotNull
+    private String thing;
+
+    @ConfigurationInject
+    public AbstractConfigImpl(@Parameter("name") @NotNull String name, @NotNull String value) {
+        super(value);
+        this.name = name;
+        this.childThing = "def childThing";
+        this.thing = "thing";
+    }
+
+    @NotNull
+    public final String getName() {
+        return this.name;
+    }
+
+    @Nullable
+    public final String getChildThing() {
+        return this.childThing;
+    }
+
+    public final void setChildThing(@Nullable String childThing) {
+        this.childThing = childThing;
+    }
+
+    @Override
+    @NotNull
+    public String getThing() {
+        return this.thing;
+    }
+
+    public void setThing(@NotNull String thing) {
+        this.thing = thing;
+    }
+}

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/ConfigurationPropertiesInheritanceSpec.groovy
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/ConfigurationPropertiesInheritanceSpec.groovy
@@ -194,4 +194,57 @@ class ConfigurationPropertiesInheritanceSpec extends Specification {
         managers[0].is(teams[0].manager)
         managers.size() == 1
     }
+
+    void "test EachProperty with overridden abstract class method"() {
+        given:
+        ApplicationContext context = ApplicationContext.run([
+                'teams.one.value': 'My value',
+                'teams.one.not-thing': 'First not thing',
+                'teams.one.thing': 'This is the thing',
+                'teams.one.child-thing': 'My child thing',
+
+                'teams.two.value': 'My second value',
+                'teams.two.thing': 'This is the big thing',
+                'teams.two.not-thing': 'Second not thing',
+        ])
+
+        when:
+        def teamOne = context.getBean(AbstractConfigImpl, Qualifiers.byName("one"))
+        def teamTwo = context.getBean(AbstractConfigImpl, Qualifiers.byName("two"))
+
+        then:
+        teamOne.name == 'one'
+        teamOne.value == 'My value'
+        teamOne.notThing == 'First not thing'
+        teamOne.thing == 'This is the thing'
+        teamOne.childThing == 'My child thing'
+
+        teamTwo.name == 'two'
+        teamTwo.value == 'My second value'
+        teamTwo.notThing == 'Second not thing'
+        teamTwo.thing == 'This is the big thing'
+        teamTwo.childThing == 'def childThing'
+    }
+
+    void "test EachProperty with overridden non-abstract class method"() {
+        given:
+        ApplicationContext context = ApplicationContext.run([
+                'teams.one.value': 'My value',
+                'teams.one.super-value': 'My super value',
+                'teams.one.super-value-with-override': 'My super overriden value',
+                'teams.one.thing': 'This is the thing',
+                'teams.one.child-thing': 'My child thing',
+        ])
+
+        when:
+        def teamOne = context.getBean(NotAbstractConfigImpl, Qualifiers.byName("one"))
+
+        then:
+        teamOne.name == 'one'
+        teamOne.value == 'My value'
+        teamOne.superValue == 'My super value'
+        teamOne.superValueWithOverride == 'My super overriden value'
+        teamOne.thing == 'This is the thing'
+        teamOne.childThing == 'My child thing'
+    }
 }

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/NotAbstractConfig.java
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/NotAbstractConfig.java
@@ -1,0 +1,42 @@
+package io.micronaut.inject.configproperties.inheritance;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class NotAbstractConfig {
+    @NotNull
+    private final String value;
+    @Nullable
+    private String superValue;
+    @Nullable
+    private String superValueWithOverride;
+
+    public NotAbstractConfig(@NotNull String value) {
+        this.value = value;
+        this.superValue = "";
+        this.superValueWithOverride = "this is ";
+    }
+
+    @NotNull
+    public final String getValue() {
+        return this.value;
+    }
+
+    @Nullable
+    public final String getSuperValue() {
+        return this.superValue;
+    }
+
+    public final void setSuperValue(@Nullable String var1) {
+        this.superValue = var1;
+    }
+
+    @Nullable
+    public String getSuperValueWithOverride() {
+        return this.superValueWithOverride;
+    }
+
+    public void setSuperValueWithOverride(@Nullable String var1) {
+        this.superValueWithOverride = var1;
+    }
+}

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/NotAbstractConfigImpl.java
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/inheritance/NotAbstractConfigImpl.java
@@ -1,0 +1,59 @@
+package io.micronaut.inject.configproperties.inheritance;
+
+import io.micronaut.context.annotation.ConfigurationInject;
+import io.micronaut.context.annotation.EachProperty;
+import io.micronaut.context.annotation.Parameter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@EachProperty("teams")
+public class NotAbstractConfigImpl extends NotAbstractConfig {
+    @NotNull
+    private final String name;
+    @NotNull
+    private final String thing;
+    @Nullable
+    private String childThing;
+    @Nullable
+    private String superValueWithOverride;
+
+    @ConfigurationInject
+    public NotAbstractConfigImpl(@Parameter("name") @NotNull String name, @NotNull String value, @NotNull String thing) {
+        super(value);
+        this.name = name;
+        this.thing = thing;
+        this.childThing = "def childThing";
+        this.superValueWithOverride = "my defaultValue";
+    }
+
+    @NotNull
+    public final String getName() {
+        return this.name;
+    }
+
+    @NotNull
+    public final String getThing() {
+        return this.thing;
+    }
+
+    @Nullable
+    public final String getChildThing() {
+        return this.childThing;
+    }
+
+    public final void setChildThing(@Nullable String var1) {
+        this.childThing = var1;
+    }
+
+    @Override
+    @Nullable
+    public String getSuperValueWithOverride() {
+        return this.superValueWithOverride;
+    }
+
+    @Override
+    public void setSuperValueWithOverride(@Nullable String var1) {
+        this.superValueWithOverride = var1;
+    }
+}
+

--- a/inject-kotlin/src/test/groovy/io/micronaut/kotlin/processing/beans/configproperties/inheritance/ConfigurationPropertiesInheritanceSpec.groovy
+++ b/inject-kotlin/src/test/groovy/io/micronaut/kotlin/processing/beans/configproperties/inheritance/ConfigurationPropertiesInheritanceSpec.groovy
@@ -194,4 +194,57 @@ class ConfigurationPropertiesInheritanceSpec extends Specification {
         managers.size() == 1
         managers[0].is(teams[0].manager)
     }
+
+    void "test EachProperty with overridden abstract class method"() {
+        given:
+        ApplicationContext context = ApplicationContext.run([
+                'teams.one.value': 'My value',
+                'teams.one.not-thing': 'First not thing',
+                'teams.one.thing': 'This is the thing',
+                'teams.one.child-thing': 'My child thing',
+
+                'teams.two.value': 'My second value',
+                'teams.two.thing': 'This is the big thing',
+                'teams.two.not-thing': 'Second not thing',
+        ])
+
+        when:
+        def teamOne = context.getBean(AbstractConfigImpl, Qualifiers.byName("one"))
+        def teamTwo = context.getBean(AbstractConfigImpl, Qualifiers.byName("two"))
+
+        then:
+        teamOne.name == 'one'
+        teamOne.value == 'My value'
+        teamOne.notThing == 'First not thing'
+        teamOne.thing == 'This is the thing'
+        teamOne.childThing == 'My child thing'
+
+        teamTwo.name == 'two'
+        teamTwo.value == 'My second value'
+        teamTwo.notThing == 'Second not thing'
+        teamTwo.thing == 'This is the big thing'
+        teamTwo.childThing == 'def childThing'
+    }
+
+    void "test EachProperty with overridden non-abstract class method"() {
+        given:
+        ApplicationContext context = ApplicationContext.run([
+                'teams.one.value': 'My value',
+                'teams.one.super-value': 'My super value',
+                'teams.one.super-value-with-override': 'My super overriden value',
+                'teams.one.thing': 'This is the thing',
+                'teams.one.child-thing': 'My child thing',
+        ])
+
+        when:
+        def teamOne = context.getBean(NotAbstractConfigImpl, Qualifiers.byName("one"))
+
+        then:
+        teamOne.name == 'one'
+        teamOne.value == 'My value'
+        teamOne.superValue == 'My super value'
+        teamOne.superValueWithOverride == 'My super overriden value'
+        teamOne.thing == 'This is the thing'
+        teamOne.childThing == 'My child thing'
+    }
 }

--- a/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/AbstractConfig.kt
+++ b/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/AbstractConfig.kt
@@ -1,0 +1,8 @@
+package io.micronaut.kotlin.processing.beans.configproperties.inheritance
+
+abstract class AbstractConfig(
+    val value: String = ""
+) {
+    var notThing: String = "def notThing"
+    abstract val thing: String
+}

--- a/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/AbstractConfigImpl.kt
+++ b/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/AbstractConfigImpl.kt
@@ -1,0 +1,15 @@
+package io.micronaut.kotlin.processing.beans.configproperties.inheritance
+
+import io.micronaut.context.annotation.ConfigurationInject
+import io.micronaut.context.annotation.EachProperty
+import io.micronaut.context.annotation.Parameter
+
+@EachProperty(value = "teams")
+open class AbstractConfigImpl @ConfigurationInject constructor(
+    @Parameter("name") val name: String,
+    value: String
+) : AbstractConfig(value = value) {
+
+    var childThing: String? = "def childThing"
+    override var thing: String = "thing"
+}

--- a/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/NotAbstractConfig.kt
+++ b/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/NotAbstractConfig.kt
@@ -1,0 +1,8 @@
+package io.micronaut.kotlin.processing.beans.configproperties.inheritance
+
+open class NotAbstractConfig(
+    val value: String = "",
+) {
+    var superValue: String? = ""
+    open var superValueWithOverride: String? = "this is "
+}

--- a/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/NotAbstractConfigImpl.kt
+++ b/inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/inheritance/NotAbstractConfigImpl.kt
@@ -1,0 +1,16 @@
+package io.micronaut.kotlin.processing.beans.configproperties.inheritance
+
+import io.micronaut.context.annotation.ConfigurationInject
+import io.micronaut.context.annotation.EachProperty
+import io.micronaut.context.annotation.Parameter
+
+@EachProperty(value = "teams")
+open class NotAbstractConfigImpl @ConfigurationInject constructor(
+    @Parameter("name") val name: String,
+    value: String,
+    val thing: String = "thing"
+) : NotAbstractConfig(value = value) {
+
+    var childThing: String? = "def childThing"
+    override var superValueWithOverride:String? = "my defaultValue"
+}

